test/cpp/end2end/end2end_test.cc

```diff
@@ -2147,13 +2147,14 @@ TEST_P(SecureEnd2endTest, ClientAuthContext) {
   std::vector<grpc::string_ref> tst =
       auth_ctx->FindPropertyValues("transport_security_type");
   ASSERT_EQ(1u, tst.size());
-  std::string credentials_type = GetParam().credentials_type;
+  std::string transport_security_type = GetParam().credentials_type;
```

Comment:
WDYT about adding a `GetTransportSecurityType` API to `test_credentials_provider.h` that converts between the credentials type and the transport security type (i.e. the thing from the AuthContext property)?

This would remove this logic from the tests and avoid duplicating code.